libpysal/graph/_vertex_set_contiguity.py

```diff
@@ -0,0 +1,102 @@
+import itertools as it
+import collections
+
+import pygeos
+from numba import njit, prange
+import numpy
+
+QUEEN = 1
+ROOK = 2
+
+__author__ = "Jay Laura jlaura@asu.edu, Levi John Wolf levi.john.wolf@gmail.com"
+
+
+class ContiguityWeightsLists:
+    """
+    Contiguity for a collection of polygons using high performance
+    list, set, and dict containers
+    """
+
+    def __init__(self, collection, wttype=1):
+        """
+        Arguments
+        =========
+
+        collection: PySAL PolygonCollection
+
+        wttype: int
+                1: Queen
+                2: Rook
+        """
+        self.n = len(collection)
+        self.coordinates = pygeos.get_coordinates(collection)
+        self.coords_per_geom = pygeos.get_num_coordinates(collection)
+        self.coord_indices = numpy.repeat(numpy.arange(self.n), self.coords_per_geom)
+        self.edge_indices = numpy.repeat(numpy.arange(self.n), self.coords_per_geom - 1)
+        self.wttype = wttype
+        self.contiguity()
+
+    def contiguity(self):
+
+        if self.wttype == QUEEN:  # points are hashed
+            hashes = _point_hash(self.coordinates)
+            indices = self.coord_indices
+        elif self.wttype == ROOK:  # edges are hashed
+            hashes = _edge_hash(self.coordinates)
+            indices = self.edge_indices
+        else:
+            raise Exception("Weight type {} Not Understood!".format(self.wttype))
+
+        hash_to_geom_lut = collections.defaultdict(set)
+        for i, hash_ in enumerate(hashes):
+            hash_to_geom_lut[hash_].add(indices[i])
+
+        edgelist = set()
+        shared_vertices = []
+        for hash_, neighbors in hash_to_geom_lut.items():
+            if len(neighbors) > 1:
+                for i in neighbors:
+                    for j in neighbors:
+                        if i != j:
+                            edgelist.update([(i, j)])
+        self.edgelist = edgelist
+
+
+@njit
+def _point_hash(coordinates):
+    n = coordinates.shape[0]
+    result = numpy.ones(n)
+    for i in range(n):
+        target = (coordinates[i, 0], coordinates[i, 1])
+        result[i] = hash(target)
+    return result
+
+
+@njit
+def _edge_hash(coordinates, indices, n_coordinates_per_geom):
+    n_coordinates = coordinates.shape[0]
+    n_edges = numpy.sum(n_coordinates_per_geom - 1)
+    result = numpy.ones(n_edges)
+    offsets = numpy.cumsum(n_coordinates_per_geom)
+    on_point = 0
+    edge_ix = 0
+    for i in range(n_coordinates):
+        on_point += 1
+        index = indices[i]
+        offset = offsets[i]
+        tail = coordinates[i]
+        if on_point == n_coordinates_per_geom[index]:  # would modulo be faster?
+            on_point = 0
+            continue
+        else:
+            head = coordinates[i + 1]
+
+        # ensure left-to-right ordering to avoid reversability issues
+        if head[0] > tail[0]:
+            tmp = head
+            head = tail
+            tail = tmp
```

Comment:
```suggestion
            head, tail = tail, head
```
bit shorter version of the same